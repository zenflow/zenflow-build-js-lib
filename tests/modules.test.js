/* eslint-env jest */

const path = require('path')
const buildFixture = require('./util/buildFixture')
const { readFile, fileExists } = require('./util/fs')
const { flatmap, repeat } = require('./util/util')

describe('fixtures/modules', () => {
  let tempDir
  beforeAll(async () => {
    tempDir = await buildFixture('modules', { extras: true })
  })
  test('snapshot (A) evaluated cjs', async () => {
    const file = path.join(tempDir, 'dist/modules.cjs.js')
    expect(require(file)).toMatchSnapshot()
  })
  test('snapshot (A) evaluated umd', async () => {
    const file = path.join(tempDir, 'dist/modules.umd.js')
    expect(require(file)).toMatchSnapshot()
  })
  test('snapshot (B) built cjs', async () => {
    const file = path.join(tempDir, 'dist/modules.cjs.js')
    expect(await readFile(file)).toMatchSnapshot()
  })
  test('snapshot (B) built es', async () => {
    const file = path.join(tempDir, 'dist/modules.es.js')
    expect(await readFile(file)).toMatchSnapshot()
  })
  test('snapshot (B) built umd', async () => {
    const file = path.join(tempDir, 'dist/modules.umd.js')
    expect(await readFile(file)).toMatchSnapshot()
  })
  test('minified versions and source maps exist in each format', async () => {
    const files = flatmap(['cjs', 'es', 'umd'], format =>
      flatmap([false, true], minify =>
        flatmap([false, true], sourcemap =>
          path.join(
            tempDir,
            'dist',
            `modules.${format}` +
              (minify ? '.min' : '') +
              '.js' +
              (sourcemap ? '.map' : ''),
          ),
        ),
      ),
    )
    const filesExist = await Promise.all(files.map(fileExists))
    expect(filesExist).toEqual(repeat(true, 3 * 2 * 2))
  })
})
