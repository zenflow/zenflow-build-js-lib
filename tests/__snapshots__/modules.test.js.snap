// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures/modules snapshot (A) evaluated cjs 1`] = `2.5000000000000004`;

exports[`fixtures/modules snapshot (A) evaluated umd 1`] = `2.5000000000000004`;

exports[`fixtures/modules snapshot (B) built cjs 1`] = `
"/** @preserve
  * package: modules v0.0.0
  * file: dist/modules.cjs.js
  * homepage: HOMEPAGE
  * license: LICENSE
  **/
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var SQRT2 = _interopDefault(require('const-sqrt-two'));

// sqrt(1/2)

var SQRT_HALF = 7.07106781186547524400844362104849039284835937688474036588339868995366239231053519425193767163820786367506923115e-01;

// EXPORTS //

var lib = SQRT_HALF;

var square = (function (number) {
  return number * number;
});

var index = square(lib) + square(SQRT2);

module.exports = index;
//# sourceMappingURL=modules.cjs.js.map
"
`;

exports[`fixtures/modules snapshot (B) built es 1`] = `
"/** @preserve
  * package: modules v0.0.0
  * file: dist/modules.es.js
  * homepage: HOMEPAGE
  * license: LICENSE
  **/
import SQRT2 from 'const-sqrt-two';

// sqrt(1/2)

var SQRT_HALF = 7.07106781186547524400844362104849039284835937688474036588339868995366239231053519425193767163820786367506923115e-01;

// EXPORTS //

var lib = SQRT_HALF;

var square = (function (number) {
  return number * number;
});

var index = square(lib) + square(SQRT2);

export default index;
//# sourceMappingURL=modules.es.js.map
"
`;

exports[`fixtures/modules snapshot (B) built umd 1`] = `
"/** @preserve
  * package: modules v0.0.0
  * file: dist/modules.umd.js
  * homepage: HOMEPAGE
  * license: LICENSE
  **/
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? module.exports = factory(require('const-sqrt-two')) :
	typeof define === 'function' && define.amd ? define(['const-sqrt-two'], factory) :
	(global._false_ = factory(global._false_));
}(this, (function (SQRT2) { 'use strict';

	SQRT2 = SQRT2 && SQRT2.hasOwnProperty('default') ? SQRT2['default'] : SQRT2;

	// sqrt(1/2)

	var SQRT_HALF = 7.07106781186547524400844362104849039284835937688474036588339868995366239231053519425193767163820786367506923115e-01;

	// EXPORTS //

	var lib = SQRT_HALF;

	var square = (function (number) {
	  return number * number;
	});

	var index = square(lib) + square(SQRT2);

	return index;

})));
//# sourceMappingURL=modules.umd.js.map
"
`;
